pdfbox/src/main/java/org/apache/pdfbox/rendering/PageDrawer.java

```diff
@@ -652,18 +674,26 @@ private BasicStroke getStroke()
     @Override
     public void strokePath() throws IOException
     {
+        checkColors();
+
         graphics.setComposite(getGraphicsState().getStrokingJavaComposite());
         graphics.setPaint(getStrokingPaint());
         graphics.setStroke(getStroke());
         setClip();
-        //TODO bbox of shading pattern should be used here? (see fillPath)
-        graphics.draw(linePath);
+
+        if (shouldDraw(getGraphicsState().getStrokingColor())) {            

```

Comment:
The purpose of still doing the sets is to maintain a similar graphics state as if we had drawn the line. This is in case other operations rely on the state being set.